rust/lance-index/src/scalar/expression.rs

```diff
@@ -1007,61 +1007,62 @@ fn visit_comparison(
     }
 }
 
-fn maybe_between(expr: &BinaryExpr) -> Option<Between> {
-    let left_comparison = match expr.left.as_ref() {
+fn maybe_range(
+    expr: &BinaryExpr,
+    index_info: &dyn IndexInformationProvider,
+) -> Option<IndexedExpression> {
+    let left_expr = match expr.left.as_ref() {
         Expr::BinaryExpr(binary_expr) => Some(binary_expr),
         _ => None,
     }?;
-    let right_comparison = match expr.right.as_ref() {
+    let right_expr = match expr.right.as_ref() {
         Expr::BinaryExpr(binary_expr) => Some(binary_expr),
         _ => None,
     }?;
 
-    match (left_comparison.op, right_comparison.op) {
+    let (left_col, dt, parser) = maybe_indexed_column(&left_expr.left, index_info)?;
+    let right_col = maybe_column(&right_expr.left)?;
+
+    if left_col != right_col {
+        return None;
+    }
+
+    let left_value = maybe_scalar(&left_expr.right, dt)?;
+    let right_value = maybe_scalar(&right_expr.right, dt)?;
```

Comment:
This assumes we always have `column op literal` and never `literal op column` but I think that's the way the DF normalizer works so we are ok.